lib/dao.js

```diff
@@ -590,18 +598,19 @@ DataAccessObject.findById = function find(id, options, cb) {
       options = {};
     }
   }
-  cb = cb || noCallback;
+  cb = cb || utils.createPromiseCallback();
   options = options || {};
 
   assert(typeof options === 'object', 'The options argument must be an object');
   assert(typeof cb === 'function', 'The cb argument must be a function');
 
   if (id == null || id === '') {
-    return process.nextTick(function() {
+    process.nextTick(function() {
       cb(new Error('Model::findById requires the id argument'));
     });
   }
-  this.findOne(byIdQuery(this, id), options, cb);
+  else this.findOne(byIdQuery(this, id), options, cb);
```

Comment:
Coding style: open a new block and put `this.findOne` on a new line. `else` is on the same line as the closing brace of the previous block.

```js
} else {
  this.findOne(...);
}
```